### ISSUE TITLE
[Feature Request][Interpolated Strings] Add option to automatic add a $ when write a "

### ISSUE THREAD

**Issue by TobiasSekan** [dotnet/roslyn#61556]
Hello,

Most time i use interpolated strings, so it would be helpful when there were a option, so that a user can activate a automatic added `$` (Dollar sign) when write a `"` (Quote), after a `=` (equals sign) and a `(` (bracket). Maybe with different options, like only after equals sign, only after brackets, etc.

Example: 
`var myText =`

Result when write a `"` (Quote) 

New option off:
`var myText = ""`

New option on:
`var myText = $""`

**Comment by svick** [comment]
Most strings in most programs shouldn't be interpolated strings, so this doesn't seem particularly useful to me, even as an option that's off by default.

Though maybe it could work with some heuristic that detects interpolated strings. Just typing `{` inside a string is probably not good enough (you could just be typing some JSON). But when you're done typing a string and it turns out it would be a valid interpolated string, then automatically adding the `$` sounds fairly reasonable to me.

**Comment by Neme12** [comment]
I agree that most string literals are actually not interpolated, so this doesn't make sense to me to apply to all string literals. But I get the sentiment that it can be annoying that when you realize you want to add an interpolation hole to your string, you have to go all the way to the beginning of the string and type something there. It's a bit of a shame that C# doesn't have any kind of string literal where interpolation is always allowed, not even the new raw string literals work that way. I really like JavaScript's back-tick string literals, where if you realize while typing the string that you need to interpolate something, you can always do it.

**Comment by Neme12** [comment]
> It's a bit of a shame that C# doesn't have any kind of string literal where interpolation is always allowed

Of course you *can* always use `$` with all your strings, but that kind of implies that it's needed and you're actually interpolating something, so it can be a bit confusing. And tempting to remove it if you don't interpolate anything because it's an extra character.

**Comment by vatsalyaagrawal** [comment]
Duplicate of #53996
